### ISSUE TITLE
Detect buf.build's buf.yaml for auto configuration of proto roots

### ISSUE THREAD

**Issue by nvx** [devkanro/intellij-protobuf-plugin#74]
When using buf.build to build protobufs a file called `buf.yaml` will exist at the root of the proto directory.

Detecting this file to automatically use these directories as proto roots would be super handy. An option could be added to disable this behaviour if some people found it undesirable for whatever reason.

**Comment by neoeinstein** [comment]
In addition, detecting those `buf.yaml` files and `buf.lock` files. The `buf.lock` files can be interpreted to find roots for dependencies. Here's an example `buf.lock` file.

```yaml
# Generated by buf. DO NOT EDIT.
version: v1
deps:
  - remote: buf.build
    owner: envoyproxy
    repository: protoc-gen-validate
    commit: dc09a417d27241f7b069feae2cd74a0e
```

On my MacOS machine, this correlates to needing to create a protobuf root at `~/.cache/buf/v1/module/data/buf.build/envoyproxy/protoc-gen-validate/dc09a417d27241f7b069feae2cd74a0e`.

**Comment by devkanro** [comment]
Planning in `buf` branch.

**Comment by devkanro** [comment]
@nvx @neoeinstein 

Some updates here!

I am working on `buf.build` spec support for this plugin.
There are my route map:
1. buf.yaml and other configuration files code completion and documentation support.
2. Add a buf setting in build tools group to config or download the `buf` binrary.
3. Auto config protobuf roots with buf.yaml and buf.work.yaml
4. Buf action menu for exec buf commands.

You can tracking my work on [buf](https://github.com/devkanro/intellij-protobuf-plugin/tree/buf) branch.

**Comment by devkanro** [comment]
@nvx @neoeinstein 

Some updates here!

The basic buf integrated is already available.

There are some features have implemented
1. basically code completion and documentation for buf configuration files.
2. auto config protobuf import roots from buf.yaml and buf.work.yaml, it support locked deps and workspace local module.
3. pre-provided well-known protos.

There are some planned feature still in development:
1. more code completion for buf configuration files, like the module navigation and name suggestion.
2. buf actions menus for exec buf command, like build, gen or push the buf modules.

There may still be some bugs about configuration file status sync here, but it works in many cases.
Hope you guys can try the plugin and give some suggestions or find some bugs.

You can download the beta build ([IntelliJ Protobuf Language Plugin 1.5.0](https://github.com/devkanro/intellij-protobuf-plugin/files/8304647/IntelliJ.Protobuf.Language.Plugin-1.5.0.zip))

**Comment by neoeinstein** [comment]
This looks great. I just did some testing, and it looks good to me. Thanks for the quick work to bring this integration to us!

**Comment by devkanro** [comment]
@nvx @neoeinstein

Some updates here!

There are some new features have implemented:

1. annotations from buf lint.
2. buf tool window for exec buf command, view the buf module tree.

The first version of buf integration is all completed, I will release it this week if no major bug.

Download 1.5.1-EAP from [Github Release](https://github.com/devkanro/intellij-protobuf-plugin/releases/tag/v1.5.1-EAP) or [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/16422-protobuf/versions/EAP/)

**Comment by devkanro** [comment]
Closed by release [1.5.0](https://github.com/devkanro/intellij-protobuf-plugin/releases/tag/v1.5.0)
